HerPublicWebsite.BusinessLogic/IDataAccessProvider.cs

```diff
@@ -19,5 +19,5 @@ public interface IDataAccessProvider
     Task<ReferralRequestFollowUp> GetReferralFollowUpByToken(string token);
     Task<ReferralRequestFollowUp> UpdateReferralFollowUpByTokenWithWasFollowedUp(string token, bool wasFollowedUp);
     Task<Session> PersistSession(Session session);
-    Task SetJourneyComplete(int sessionId);
+    Task SetEligibilityAndJourneyComplete(int sessionId, bool? isEligibleValue);
```

Comment:
```suggestion
    Task SetEligibilityAndJourneyComplete(int sessionId, bool? isEligible);
```
Don't think `...Value` is adding anything here.

Also wonder if there's an argument for just leaving this as `SetJourneyComplete`? There's something a bit strange about `SetEligibility` in my mind here - maybe suggests we're doing something that we're not, like determining eligibility? Perhaps `RecordEligibilityAndJourneyCompletion`?

Reply:
Also maybe worth talking about - did you consider giving `isEligible` a default value?

Reply:
Yeah, this felt like something quite tricky to name! 
For the default value, I decided against it because it felt like it I wanted it to be a conscious and explicit decision when this function is used, as whatever a future dev puts in here will be recorded in the database. My thinking was that using the default value kind of hides that choice. Also as journeys ending and eligibility decisions go hand-in-hand, using null as a default didn't really seem to be useful either.

Reply:
I think the value was linked to the naming. I wanted to put the `...Value` to communicate we weren't deciding anything here, the value already existed, and it was just being entered there. But changing the name to `Record...` makes sense and makes it feel cleaner and more descriptive.

Reply:
Nice, yes: agree about forcing `isEligible` to be a conscious choice.